cli/command/service/ps.go

```diff
@@ -74,27 +74,34 @@ func runPS(dockerCli *command.DockerCli, opts psOptions) error {
 		return err
 	}
 
+loop:
 	for _, service := range opts.services {
-		serviceCount := 0
-		// Lookup by ID/Prefix
+		// Lookup by full ID match
 		for _, serviceEntry := range serviceByIDList {
-			if strings.HasPrefix(serviceEntry.ID, service) {
+			if serviceEntry.ID == service {
 				filter.Add("service", serviceEntry.ID)
-				serviceCount++
+				continue loop
 			}
 		}
 
-		// Lookup by Name/Prefix
+		// Lookup by Name
 		for _, serviceEntry := range serviceByNameList {
-			if strings.HasPrefix(serviceEntry.Spec.Annotations.Name, service) {
+			if serviceEntry.Spec.Annotations.Name == service {
 				filter.Add("service", serviceEntry.ID)
-				serviceCount++
+				continue loop
 			}
 		}
-		// If nothing has been found, return immediately.
-		if serviceCount == 0 {
-			return errors.Errorf("no such services: %s", service)
+
+		// Lookup by ID-prefix
+		for _, serviceEntry := range serviceByIDList {
+			if strings.HasPrefix(serviceEntry.ID, service) {
+				filter.Add("service", serviceEntry.ID)
+				continue loop
```

Comment:
Here we're choosing the first service whose ID matches the prefix, and ignoring the others. I think we should error out if the prefix is ambiguous. There are several places in the code where we do this, and again, I think in the long term it would be a good idea to consolidate the code in a central package.